jclklib/proxy/connect_ptp4l.cpp

```diff
@@ -120,14 +118,16 @@ void event_handle()
             if (portState <= PASSIVE) {
                 pe.master_offset = 0;
                 memset(pe.gm_identity, 0, sizeof(pe.gm_identity));
-                pe.servo_locked = 0;
+                pe.synced_to_primary_clock = false;
+            } else {
+                pe.synced_to_primary_clock = true;
             }
-
             break;
         case PORT_PROPERTIES_NP:
             /* Retrieve current port state when proxy is started */
             pd = (PORT_DATA_SET_t *)data;
             portState = pd->portState;
+            pe.synced_to_primary_clock = portState >= SLAVE ? true:false;
```

Comment:
use time receiver

Reply:
I got this from fsm.h ( linux-ptp)

/** Defines the state of a port. */
enum port_state {
	PS_INITIALIZING = 1,
	PS_FAULTY,
	PS_DISABLED,
	PS_LISTENING,
	PS_PRE_MASTER,
	PS_MASTER,
	PS_PASSIVE,
	PS_UNCALIBRATED,
	PS_SLAVE,
	PS_GRAND_MASTER, /*non-standard extension*/ 
}; 

Maybe we shd stick with SLAVE ?  to be aligned with linux-ptp naming. I understand the concern with name we shouldnt use. 


Future note : 
I wonder why libptpmgmt didnt use directly the linux-ptp header  (fsm.h)

Reply:
Just for reference:
/** Port state */
enm(portState_e) sz(: uint8_t) {
    NM(INITIALIZING)         = 1, /**< Initializing */
    NM(FAULTY)               = 2, /**< Faulty */
    NM(DISABLED)             = 3, /**< Disabled */
    NM(LISTENING)            = 4, /**< Listening */
    NM(PRE_MASTER)           = 5, /**< Pre timeTransmitter */
    NM(PRE_TIME_TRANSMITTER) = 5, /**< Pre timeTransmitter */
    NM(MASTER)               = 6, /**< TimeTransmitter */
    NM(TIME_TRANSMITTER)     = 6, /**< TimeTransmitter */
    NM(PASSIVE)              = 7, /**< Passive */
    NM(UNCALIBRATED)         = 8, /**< Uncalibrated */
    NM(SLAVE)                = 9, /**< TimeReceiver */
    NM(TIME_RECEIVER)        = 9, /**< TimeReceiver */
};